src/cloud/docker/docker-test-magecloud-pkg-code.md

```diff
@@ -0,0 +1,263 @@
+
+1. Add the following hostname to your `/etc/hosts` file.
+
+   ```conf
+   127.0.0.1 magento2.docker
+   ```
+
+   Alternatively, you can run the following command to add it to the file:
+
+   ```bash
+   echo "127.0.0.1 magento2.docker" | sudo tee -a /etc/hosts
+   ```
+
+1. Add required project dependencies if you are testing the following packages.
+
+   -  For the `{{site.data.var.mcp-package}}` package:
+
+      ```bash
+      composer require "magento/magento-cloud-docker:^1.0.0" --no-update
+      ```
+
+   -  For the `{{site.data.var.mcc-package}}` package:
+
+      ```bash
+      composer require "magento/magento-cloud-docker:^1.0.0" --no-update
+      composer require "magento/framework:*" --no-update
+      composer require "magento/module-store:*" --no-update
+      composer require "magento/module-url-rewrite:*" --no-update
+      ```
+
+1. Switch to the preferred PHP version for running tests.
+
+1. Update the project dependencies.
+
+   ```bash
+   composer update
+   ```
+
+### Add credentials
+
+Add credentials to your Docker environment using any of the following methods:
+
+-  Use environment variables
+-  Load credentials from the environment configuration file
+-  Add variables directly to the test configuration file
+
+{:.procedure}
+To add credentials using environment variables:
+
+1. Create environment variables for your GitHub authentication keys.
+
+   ```bash
+   export REPO_USERNAME=your_public_key
+   ```
+
+   ```bash
+   export REPO_PASSWORD=your_private_key
+   ```
+
+1. Some packages require a GitHub authentication during installation. Create an environment variable with your GitHub token that can be used to install these packages.
+
+   ```bash
+   `export GITHUB_TOKEN=your_github_token
+   ````
+
+{:.procedure}
+To load credentials from the environment configuration file:
+
+1. Run the following commands to write credentials to the ./.env file.
+
+   ```bash
+   echo "REPO_USERNAME=your_public_key" >> ./.env
+   ```
+
+   ```bash
+   echo "REPO_PASSWORD=your_private_key" >> ./.env
+   ```
+
+1. Edit the `codeception.dist.yml` file.
+
+   ```yaml
+   params:
+       - tests/functional/configuration.dist.yml
+       - env
+       - .env
+   ```
+
+1. Add a dependency for the [vlucas/phpdotenv] package required to load the environment variables.
+
+   ```bash
+   composer require "vlucas/phpdotenv": "^3.0"
+   ```
+
+{:.procedure}
+To add credentials directly to the test configuration file:
+
+1. Open the `codeception.dist.yml` file in an editor.
+
+1. Replace the `%REPO_USERNAME%`, `%REPO_PASSWORD%`, and `%GITHUB_TOKEN%` placeholder values with your credentials:
+
+   ```yaml
+   modules:
+       config:
+           Magento\CloudDocker\Test\Functional\Codeception\TestInfrastructure:
+               ...
+               composer_magento_username: "%REPO_USERNAME%"
+               composer_magento_password: "%REPO_PASSWORD%"
+               composer_github_token: "%GITHUB_TOKEN%"
+               ...
+   ```
```

Comment:
For some reason, this is not showing well in the preview. I wonder if it is because of the ellipsis...

Reply:
The ellipsis are OK in the following example. YAML validator gives ```(<unknown>): mapping values are not allowed in this context at line 5 column 35```

Will work on fixing after release.